content/blog/2016/2016-05-25-gsoc-jenkins-web-ui-project.adoc

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: "GSoC Project: Improving Job Creation and Configuration"
+- core
+- gsoc
+author: samatdav
+---
+
+=== About me
+My name is Samat and I am from Moscow, Russia. I interned at Intel and Yandex, and cofounded http://expfood.ru/[a startup project] where I personally developed front-end and back-end of the website.
+I am excited to participate in GSoC with Jenkins this summer as a chanсe to make a positive change for thousands of users as well as to learn from great mentors.
+
+=== Abstract
+Although powerful, Jenkins new job creation and configuration process may be non obvious and time consuming. This can be improved by making UI more intuitive, concise, and functional. I plan to achieve this by creating a simpler new job creation, configuration process focused on essential elements, and embedding new functionality. 
+
+My mentors are https://github.com/lanwen[Kirill] and https://github.com/michaelneale[Michael]
+
+=== Deliverables
+==== New job creation
+* New job name validation
+
+Initially, job validation was unresponsive, job creation was still allowed with an invalid name, and some allowed characters even crashed Jenkins. Happily, two of this problems were fixed in recent improvements and I plan add only a real time name check for invalid characters.
+
+* Popup window. 
+
+image::http://i.imgur.com/6O3jcuA.gif[Popup window, 600]
+
+Jenkins has a lot of windows reloads that may time consuming. The creation of new job is a simple process requiring only job name and job type. This way UI may be improving by reducing page reloads and putting new job creation interface in a dialog window. Such popup would likely consist of three steps of implementation: rendering a dialog window, receiving JSON with job types, sending a POST request to create the job.
```

Comment:
(Advice: you possibly want to hit enter between sentences, or meaningful consistent blocks. This helps make the diff smaller when reviewing/changing code.)

> This way UI may be improving by reducing 

=> improved, you mean?